conda/core/solve.py

```diff
@@ -219,7 +220,14 @@ def solve_final_state(self, deps_modifier=NULL, prune=NULL, ignore_pinned=NULL,
         for pkg_name, spec in iteritems(specs_map):
             matches_for_spec = tuple(dist for dist in solution if spec.match(index[dist]))
             if matches_for_spec:
-                assert len(matches_for_spec) == 1
+                if len(matches_for_spec) != 1:
+                    raise CondaError(dals("""
+                    Conda encountered an error with your environment.  Please report an issue
+                    at https://github.com/conda/conda/issues/new.  In your report, please include
+                    the output of 'conda info' and 'conda list' for the active environment, along
+                    with the command you invoked that resulted in this error.
+                      matches_for_spec: %s
+                    """) % matches_for_spec)
```

Comment:
Looks good, but don't we have almost the exact same text somewhere else? Does it make sense to reuse it instead of rewriting?

Reply:
We do, except it's only displayed for exceptions that don't subclass `CondaError`.  This in particular is a `CondaError`, which means the full stack trace won't be shown (which is what the intention is here).

This exception is also meant to be temporary.  It'll be removed as soon as we understand the root cause of the issue.

Note that I pulled #6430 from our auto-reports.  There was no user who created an issue for us to interact with.

Reply:
Got it!